vcmi/vcmi
# Mods versioning [part 3]

Those changes were developed in parallel with develop, so I had to resolve ~100 conflicts to merge everything.
Basically, change have 3 big topics:
 - VLC for obstacles
 - Mod versioning solution (without Launcher part) - that's required for VLC for obstacles as it breaks not only backward compatibility for saves, but also all mods with new obstacles become incompatible
 - Some features developed by me and @DjWarmonger for further support of editor, in particular
 > progress bar to reflect loading progress in parallel thread (can be used to show loading bar in game during rmg computations)
> undo/redo implementation for map
> filesystem missing functions

Test scenarios to check mod versions
1. run game regular -> save game -> load game
2. run game with mod A -> save game -> load game -> load game from p1
3. run game with mod B -> save game -> load game -> load game from p1, p2
4. run game with next version of A, B -> save game -> load game from p2, p3 (error expected)
5. run game as in p.3 -> load game from p4 (error expected)
